build_catalog.sh

```diff
@@ -1,105 +1,186 @@
 
-[ $count -gt 0 ] && exit 1
-
-num_commits=$(git rev-list $(git rev-list --max-parents=0 HEAD)..HEAD --count)
-current_commit=$(git rev-parse --short=7 HEAD)
-version="0.1.$num_commits-$current_commit"
-opm_version="1.14.0"
-
-# Login to docker
-docker_conf="$PWD/.docker"
-mkdir -p "$docker_conf"
-docker_cmd="docker --config=$docker_conf"
-
-$docker_cmd login -u="$QUAY_USER" -p="$QUAY_TOKEN" quay.io
-
-# Find the CSV version from the previous bundle
-log "Pulling latest bundle image $BUNDLE_IMAGE"
-$docker_cmd pull $BUNDLE_IMAGE:latest && exists=1 || exists=0
-
-if [ $exists -eq 1 ]; then
-  log "Extracting previous version from bundle image"
-  docker create --name="tmp_$$" $BUNDLE_IMAGE:latest sh
-  tmp_dir=$(mktemp -d -t sa-XXXXXXXXXX)
-  pushd $tmp_dir
-    docker export tmp_$$ | tar -xf -
-    prev_version=`find . -name *.clusterserviceversion.* | xargs cat - | python3 -c 'import sys,yaml; print(yaml.safe_load(sys.stdin.read())["metadata"]["name"])'`
-    if [[ "$prev_version" == "" ]]; then
-      log "Unable to find previous bundle version"
-      exit 1
+[[ $count -gt 0 ]] && exit 1
+
+# We shouln't need set -u as we have checked for the vars before
+# but let's play safe
+set -euo pipefail
+
+# General vars to modify behaviour of the script
+DRY_RUN=${DRY_RUN:-true}
+DELETE_TEMP_DIR=${DELETE_TEMP_DIR:-true}
+REMOVE_UNDEPLOYED=${REMOVE_UNDEPLOYED:-false}
+
+temp_dir=$(mktemp -d)
+[[ "$DELETE_TEMP_DIR" == "true" ]] && trap 'rm -rf $temp_dir' EXIT
+
+engine_cmd="$CONTAINER_ENGINE --config=$CONFIG_DIR"
+
+# clone bundle repo containing current version
+saas_operator_dir_base="$temp_dir/saas-operator-dir"
+bundle_versions_file="$saas_operator_dir_base/$OPERATOR_NAME/${OPERATOR_NAME}-versions.txt"
+bundle_versions_package_file="$saas_operator_dir_base/$OPERATOR_NAME/${OPERATOR_NAME}.package.yaml"
+
+log "Cloning $BUNDLE_VERSIONS_REPO"
+if [[ -n "${APP_SRE_BOT_PUSH_USER:-}" && "${APP_SRE_BOT_PUSH_TOKEN:-}" ]]; then
+    bundle_versions_repo_url="https://${APP_SRE_BOT_PUSH_USER}:${APP_SRE_BOT_PUSH_TOKEN}@$BUNDLE_VERSIONS_REPO"
+else
+    bundle_versions_repo_url="https://$BUNDLE_VERSIONS_REPO"
+fi
+
+git clone --branch "$BRANCH_CHANNEL" "$bundle_versions_repo_url" "$saas_operator_dir_base"
+
+prev_operator_version=""
+removed_versions=""
+if [[ -f "$bundle_versions_file" ]]; then
+    log "$bundle_versions_file exists. We'll use to determine current version"
+    if [[ "$REMOVE_UNDEPLOYED" == "true" ]]; then
+        log "Checking if we have to remove any versions more recent than deployed hash"
+        # TODO: Move this out of here
+        deployed_hash=$(
+            curl -s "https://gitlab.cee.redhat.com/service/app-interface/-/raw/master/data/services/deployment-validation-operator/cicd/saas.yaml" | \
+                docker run --rm -i quay.io/app-sre/yq -r '.resourceTemplates[]|select(.name="deployment-validation-operator").targets[]|select(.namespace["$ref"]=="/openshift/app-sre-stage-01/namespaces/app-sre-dvo-per-cluster.yml")|.ref'
```

Comment:
This could return master, right?  Wouldn't there need to be a way to determine the actual sha?